src/main/redux/sagas/telemetry.ts

```diff
@@ -0,0 +1,122 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import * as debug_ from "debug";
+import { _APP_VERSION } from "readium-desktop/preprocessor-directives";
+import { call, select } from "typed-redux-saga";
+import { RootState } from "../states";
+import { version as osVersion } from "os";
+import * as fs from "fs";
+import * as path from "path";
+import { app } from "electron";
+
+interface ITelemetryInfo {
+    os_version: string;
+    locale: string;
+    timestamp: string;
+    fresh: boolean;
+    type: string;
+    actual_version: string;
+    prev_version: string;
+};
+
+// Logger
+const filename_ = "readium-desktop:main:redux:sagas:telemetry";
+const debug = debug_(filename_);
+
+// set info to a file queue.
+const userDataPath = app.getPath("userData");
+const folderPath = path.join(
+    userDataPath,
+    "telemetry",
+);
+if (!fs.existsSync(userDataPath)) {
+    fs.mkdirSync(userDataPath);
+}
+if (!fs.existsSync(folderPath)) {
+    fs.mkdirSync(folderPath);
+}
+const QUEUE_FILENAME = "queue.json";
+const queueFilePath = path.join(
+    folderPath,
+    QUEUE_FILENAME,
+);
+
+const EMPTY_ARRAY = "[]";
+let dataFromFileQueue = EMPTY_ARRAY;
+try {
+    if (!fs.existsSync(queueFilePath)) {
+        fs.writeFileSync(queueFilePath, dataFromFileQueue, { encoding: "utf8" });
+    } else {
+        dataFromFileQueue = fs.readFileSync(queueFilePath, { encoding: "utf-8" });
+    }
+} catch {
+    // ignore
+}
+
+function* collectAndSave() {
+
+    const version = yield* select((state: RootState) => state.version);
+
+    // fresh install is equal to en language ?
+    const locale = yield* select((state: RootState) => state.i18n.locale);
+
+    let fresh = false;
+    if (_APP_VERSION !== version) {
```

Comment:
could we please add a comment here to explain that:

yield call(telemetry.collectSaveAndSend);

is invoked before:

 yield put(appActions.initSuccess.build());

...which is essential for the version comparison to work.

thanks :)

Reply:
done